crates/corro-agent/src/agent/uni.rs

```diff
@@ -0,0 +1,172 @@
+                let rx = tokio::select! {
+                    rx_res = conn.accept_uni() => match rx_res {
+                        Ok(rx) => rx,
+                        Err(e) => {
+                            debug!("could not accept unidirectional stream from connection: {e}");
+                            return;
+                        }
+                    },
+                    _ = &mut tripwire => {
+                        debug!("connection cancelled");
+                        return;
+                    }
+                };
+
+                increment_counter!("corro.peer.stream.accept.total", "type" => "uni");
+
+                debug!(
+                    "accepted a unidirectional stream from {}",
+                    conn.remote_address()
+                );
+
+                tokio::spawn({
+                    let process_uni_tx = process_uni_tx.clone();
+                    async move {
+                        let mut framed = FramedRead::new(rx, LengthDelimitedCodec::new());
+
+                        loop {
+                            match StreamExt::next(&mut framed).await {
+                                Some(Ok(b)) => {
+                                    counter!("corro.peer.stream.bytes.recv.total", b.len() as u64, "type" => "uni");
+                                    match UniPayload::read_from_buffer(&b) {
+                                        Ok(payload) => {
+                                            trace!("parsed a payload: {payload:?}");
+
+                                            if let Err(e) = process_uni_tx.send(payload).await {
+                                                error!(
+                                                    "could not send UniPayload for processing: {e}"
+                                                );
+                                                // this means we won't be able to process more...
+                                                return;
+                                            }
+                                        }
+                                        Err(e) => {
+                                            error!("could not decode UniPayload: {e}");
+                                            continue;
+                                        }
+                                    }
+                                }
+                                Some(Err(e)) => {
+                                    error!("decode error: {e}");
+                                }
+                                None => break,
+                            }
+                        }
+                    }
+                });
+            }
+        }
+    });
+}
+
+/// Start an async buffer task that pull messages from one channel
+/// (process_uni_rx) and moves them into the next if the variant
+/// matches (bcast_msg_tx)
+// DOCME: why is this task needed?  Do we just want to get this
+// function out of the stream hot path?
```

Comment:
It is possibly not needed. I think it used to do more, but not so much anymore!

However, we definitely don't want to slow down receiving broadcasts in any way. We could spawn tasks for each broadcast we get, this is likely more conservative with resources.